File: python/day1.py
from typing import List

from utils import ISolution


class Day1Solution(ISolution):

    _DAY_STRING: str = "1"

    @classmethod
    def _parse_input(cls, raw_input: str) -> List[List[int]]:
        raw_elves = raw_input.split("\n\n")
        parsed_input = []
        for elf in raw_elves:
            parsed_input.append([int(raw_cal) for raw_cal in elf.split("\n")])
        return parsed_input

    @classmethod
    def __get_totals(cls, elves: List[List[int]]) -> List[int]:
        return [sum(elf) for elf in elves]

    @classmethod
    def _part_one(cls, problem_input) -> int:
        return max(cls.__get_totals(problem_input))

    @classmethod
    def _part_two(cls, problem_input) -> int:
        return sum(sorted(cls.__get_totals(problem_input), reverse=True)[:3])


if __name__ == "__main__":
    print("===SAMPLE===")
    Day1Solution.do_solution()
    print("===ACTUAL===")
    Day1Solution.do_solution(False)

File: nodejs/day1.ts
import { ISolution } from './ISolution'
import Utils from './utils';
type Day1Input = number[][]
type Day1Output = number
export class Day1Solution extends ISolution<Day1Input, Day1Output> {
    private totals: number[];
    protected parseRawInput(rawInput: string): Day1Input {
        let rawElves = rawInput.split("\n\n");
        let parsedInput = [];
        for (const elf of rawElves) {
            parsedInput.push(elf.split("\n").map(x => parseInt(x)));
        }
        return parsedInput;
    }

    private calcTotals(input: Day1Input) {
        this.totals = input.map(elf => Utils.sum(elf));
    }
    protected solvePartOne(input: Day1Input): Day1Output {
        this.calcTotals(input);
        return Math.max(...this.totals);
    }
    protected solvePartTwo(input: Day1Input): Day1Output {
        let topThree = Utils.sorted(this.totals).slice(-3);
        return Utils.sum(topThree);
    }
}